### DISCUSSION TITLE
Auto open browser when run dev

### DISCUSSION THREAD

**Discussion by ricardocanelas** [vercel/next.js#13448] [Help] [answered]
Is there a way to open a browser when running `next dev`?

**Comment by jamesmosier** [comment, marked as answer]
Unfortunately that is not possible currently. See https://github.com/zeit/next.js/issues/12233

> We're not planning to add automatically opening the browser at this point (keeping the CLI minimal and features contained).

**Comment by methodbox** [comment]
For a simple work-around on a *nix system just do this in `package.json`:

```json
  "scripts": {
      "dev": "npm run open-browser && next dev",
      "open-browser": "open http://localhost:3000",
}
```

The page will load once the server starts, but it will show "can't be reached" for a moment. It's not perfect but it works fine.

**Reply by kakaponick** [reply]
for windows
```
"open-browser": "start http://localhost:3000"
```

**Reply by orassayag** [reply]
Saved my day :)

**Reply by malkiii** [reply]
I personally use this in my project for `windows`, `linux` and also `WSL`

```json
"dev": "URL=http://localhost:3000 && (open $URL || cmd.exe /c start $URL) && next dev"
```

**Reply by hmzaabasi** [reply]
Worked like a charm 🎉Thanks

**Reply by TheHalcyonSavant** [reply]
For incognito lovers without `open-browser`:
```bash
cmd.exe /c start chrome.exe -incognito http://localhost:3000 && next dev
```

**Comment by ViniciusFXavier** [comment]
`--open` flag or `-o` =D

**Reply by OscarJVD** [reply]
It doesn't works.

**Comment by OscarJVD** [comment]
![image](https://user-images.githubusercontent.com/57403552/143668091-fba3bb9c-b7c0-4fea-96af-ff6d13feb900.png)
This solution works for me. Hire me now!

**Reply by celinedrules** [reply]
This is the only thing that worked for me.

**Reply by laurencioScript** [reply]
in my windows that worked for me 👍

**Reply by thucne** [reply]
Thank you. It works for me, too.

**Reply by dayanrr91** [reply]
Great, this still works

**Reply by gusperandio** [reply]
Thanks bro! It s worked

**Reply by rafiqulshopon** [reply]
It worked. Thanks

**Reply by jaydioar** [reply]
why using that open-browser task there, it's not nescesscary just use a "pre"-script. (https://docs.npmjs.com/cli/v10/using-npm/scripts) e.g.:

```
"predev": "start http://localhost:3010",
"dev": "next dev -p 3010",

```

**Comment by kiranvj** [comment]
I use this for windows 

`"dev": "explorer http://localhost:3000 & next dev",`

Please note that there is only one `&` , this is to execute both commands in parallel 

Instead of `explorer` `start` will also work `"dev": "start http://localhost:3000/?queue=true & next dev"`, - I prefer this if you have query params in url

From windows command prompt  execute `npm run dev`. It should open default browser.

**Reply by xiaogaoxiansheng** [reply]
Thanks bro! It s worked

**Reply by abhishek5670** [reply]
it worked for me

**Comment by o-az** [comment]
I suggest you leave the existing `dev` script alone and add your own. Overall more readable and still keeping `create-next-app` "defaults." Just add these 2 lines:
```JSON
  "open-browser": "open http://0.0.0.0:3000",
  "dev-open-browser": "next dev -p 3000 & $npm_execpath open-browser"
```

`$npm_execpath` allows scripts to work with both `npm` and `yarn` (and `pnpm` too).
`$npm_execpath` is replaced with `npm` or `yarn` depending on what command ran it. 

note: for Windows you might need to replace `$npm_execpath` with `%npm_execpath%`.

**Reply by sohebcakewalk** [reply]
@o-az - Worked for me on mac
open will open url in browser but start will not in my mac
thanks once again

**Comment by ISMAILBOUADDI** [comment]
go to package.json and updated like this :
` "scripts": {
      "dev": "npm run open-browser && next dev",
      "open-browser": "start http://localhost:3000",
}`

**Reply by TechSculptorx** [reply]
yes, it is work for me thanks.

**Comment by rgurgul** [comment]
For linux
"start": "npm run open && next dev -p 3010",
"open": "xdg-open http://localhost:3010",

**Comment by narender56** [comment]
try

```
"dev": "next dev & open http://localhost:3000"
```

**Comment by phyzical** [comment]
this will avoid the need for the refresh, just pipes the open over to a background process, also avoid the ghost process stuff that some solutions suggest result in
```
"dev": "npm run open && next dev",
 "open": "sleep 5 && open http://localhost:3000 &",
 ```

**Comment by gregbarbosa** [comment]
Taking inspo from @jaydioar and @phyzical, I used a [prescript](https://docs.npmjs.com/cli/v10/using-npm/scripts#pre--post-scripts).

```
"predev": "sleep 3 && open http://localhost:3000 &",
"dev": "astro dev",
```

**Comment by simoncdna** [comment]
Hello,
Since this feature is still not supported natively by next dev, I created a small plugin that provides a simple alternative for automatically opening the browser when the development server starts. it's for webpack only.

👉 [next-plugin-open-browser](https://github.com/simoncdn/next-plugin-open-browser)

If this is something you still need, feel free to try it out or give feedback.
I hope it's helpful. Thanks!

**Comment by oviirup** [comment]
Hello developers,

I have a simpler solution using VS Code debugging. You can create a launch configuration that opens the app in a new Chrome tab. Here is the configuration:

```jsonc
// .vscode/launch.json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "debug next.js",
      "type": "node-terminal",
      "request": "launch",
      "command": "bun run dev -- --inspect",
      "serverReadyAction": {
        "action": "openExternally",
        "pattern": "- Local:.+(https?://.+)",
        "uriFormat": "%s"
      }
    }
  ]
}

```

You can read more about debugging Next.js in the [official docs](https://nextjs.org/docs/app/guides/debugging)
